superset/db_engine_specs/trino.py

```diff
@@ -283,6 +286,8 @@ def _execute(
         )
         execute_thread.start()
 
+        # Wait for the thread to start before continuing
+        time.sleep(0.1)
```

Comment:
@michael-s-molina I don't think that would work, as `.set()` is only called once the thread is completed. Should we perhaps be checking that `execute_thread.is_alive()` is returning `True` before proceeding? I think that's what we're trying to do here.

Reply:
@justinpark Could you try @villebro's suggestion?

Reply:
You might also be able to use an event `.wait` instead of `sleep`:

```python
    while not execute_event.is_set():
            if cursor.query_id:
                break
            execute_event.wait(0.1)
```